utils/src/server/resource_limiting.rs

```diff
@@ -0,0 +1,82 @@
+use std::sync::Arc;
+
+use arrayvec::ArrayVec;
+use jsonrpsee_types::error::Error;
+use parking_lot::Mutex;
+
+const RESOURCE_COUNT: usize = 8;
+
+/// Fixed size table mapping a resource to some value
+pub type ResourceTable<T> = [T; RESOURCE_COUNT];
+/// Variable size table mapping a resource to some value
+pub type ResourceVec<T> = ArrayVec<T, RESOURCE_COUNT>;
+
+/// User defined resources used by the JSON-RPC server.
```

Comment:
There is no real point in defining a resource if it's infinite, and defining `0` is kind of pointless as it would only allow executing methods that cost `0`, and then allow infinite number of those. Not sure if those edge cases need describing or whether or not they are kind of intuitive :man_shrugging:?